### PR TITLE
Add the pipe pattern section

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23948

The rendered doc page is available here https://richard-akka-doc.firebaseapp.com/futures.html

As in #23948 I am planning to update `futures.md` in multiple PRs. However, let me know if I should better do everything in a single PR.

### CODE REVIEW

File: akka-docs/src/main/paradox/futures.md [hunk 1]
```diff
@@ -91,15 +91,78 @@ When using non-blocking it is better to use the `mapTo` method to safely try to
 The `mapTo` method will return a new `Future` that contains the result if the cast was successful,
 or a `ClassCastException` if not. Handling `Exception`s will be discussed further within this documentation.
 
-@@@ 
+@@@
+
+## Use the pipe pattern
+
+Another useful message-transfer pattern is "pipe", which is to send the result of `Future` to another actor, upon completion of the `Future`.
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/futures.md]: > Another useful message-transfer pattern is

I'm going to re-write the previous section as "Use the ask pattern", that's why I said "Another ... pattern" here. The previous section would be re-written in a separate PR.

File: akka-docs/src/test/java/jdocs/future/FutureDocTest.java [hunk 2]
```diff
@@ -112,6 +120,122 @@
...
+
+    @Override
+    public Receive createReceive() {
+      return receiveBuilder()
+        .match(String.class, msg -> {
+          CompletableFuture<Object> fut =
+            ask(target, "some message", timeout).toCompletableFuture();
+
+          // the pipe pattern
+          pipe(fut, getContext().dispatcher()).to(sender());
```
Comment (patriknw) [inline akka-docs/src/test/java/jdocs/future/FutureDocTest.java]: I think we try to prefer `getSender()` in docs for Java

File: akka-docs/src/test/java/jdocs/future/FutureDocTest.java [hunk 3]
```diff
@@ -112,6 +120,122 @@
...
+        })
+        .build();
+    }
+  }
+  //#pipe-to-usage
+
+  //#pipe-to-returned-data
+  public class UserData     { final String data; UserData(String data){ this.data = data; } }
+  public class UserActivity { final String activity; UserActivity(String activity){ this.activity = activity; } }
```
Comment (patriknw) [inline akka-docs/src/test/java/jdocs/future/FutureDocTest.java]: that formatting is too weird, doesn't matter that it becomes a few lines longer

File: akka-docs/src/test/java/jdocs/future/FutureDocTest.java [hunk 4]
```diff
@@ -112,6 +120,122 @@
...
+  }
+
+  public class GetFromUserActivityActor {}
+  //#pipe-to-user-activity-actor
+
+  //#pipe-to-proxy-actor
+  public class UserProxyActor extends AbstractActor {
+    ActorRef userActor;
+    ActorRef userActivityActor;
+    Timeout timeout = new Timeout(Duration.create(5, "seconds"));
```
Comment (patriknw) [inline akka-docs/src/test/java/jdocs/future/FutureDocTest.java]: Prefer the TimeUnit API over the string based. However, nowadays we prefer `java.time.Duration` in all places for Java API. There is a `Timeout.create(java.time.Duration)`

**Review by patriknw — APPROVED** [review]
LGTM, a few small nitpicks

**Comment by richardimaoka** [comment]
Hi @patriknw, thanks for the advice. Reflected your comments.

File: akka-docs/src/main/paradox/futures.md [hunk 5]
```diff
@@ -91,15 +91,82 @@ When using non-blocking it is better to use the `mapTo` method to safely try to
 The `mapTo` method will return a new `Future` that contains the result if the cast was successful,
 or a `ClassCastException` if not. Handling `Exception`s will be discussed further within this documentation.
 
-@@@ 
+@@@
+
+## Use the pipe pattern
+
+Another useful message-transfer pattern is "pipe", which is to send the result of @scala[`Future`]@java[`CompletableFuture`] to another actor, upon completion of the @scala[`Future`]@java[`CompletableFuture`].
+The pipe pattern can be used by importing @java[`akka.pattern.PatternsCS.pipe`.]@scala[`akka.pattern.pipe`, and define or import an implicit instance of `ExecutionContext` in the scope.]
```
Comment (raboof) [inline akka-docs/src/main/paradox/futures.md]: 'define or import' -> 'defining or importing'?

**Review by raboof — APPROVED** [review]

**Pull request merged**